### PR TITLE
2.11.0 release

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

Prepare 2.11.0 release

## Release Notes

Milestone: 2.11.0

Changelog: n/a

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Comment by adam-wolfe** [comment]
I am seeing an issue not seen in 2.10:

Whenever I try to expand a profile in any of the trees, I get an error message:
`Your connection is no longer active. Please log in to an authentication service to restore the connection.`

In the logs I have:
[2023/09/12 10:11:14] [ERROR] Error: Token auth error
{
  errorDetails: [ImperativeError: Token auth error
  	at Profiles.<anonymous> (c:\Users\me\GitRepos\vscode-extension-for-zowe\packages\zowe-explorer\out\src\extension.js:189073:31)
  	at Generator.next (<anonymous>)
  	at fulfilled (c:\Users\me\GitRepos\vscode-extension-for-zowe\packages\zowe-explorer\out\src\extension.js:189011:58)] {
    mDetails: {
      msg: 'Token auth error',
      additionalDetails: 'Profile was found using token auth, please log in to continue.',
      errorCode: '401'
    },
    mMessage: 'Token auth error'
  },
  label: 'dev.zosmf',
  moreInfo: 'Token auth error'
}

I am guessing that this is an issue with isUsingTokenAuth() picking up that I have used tokens in the past with certain profiles that I no longer have in my config. My current zowe.config.json only contains secure arrays containing user and password.

**Comment by JillieBeanSim** [comment]
@rudyflores can you look into this ☝🏻

**Comment by rudyflores** [comment]
> 



> I am seeing an issue not seen in 2.10:
> 
> Whenever I try to expand a profile in any of the trees, I get an error message: `Your connection is no longer active. Please log in to an authentication service to restore the connection.`
> 
> In the logs I have:
> 
> ```
> [2023/09/12 10:11:14] [ERROR] Error: Token auth error
> {
>   errorDetails: [ImperativeError: Token auth error
>   	at Profiles.<anonymous> (c:\Users\me\GitRepos\vscode-extension-for-zowe\packages\zowe-explorer\out\src\extension.js:189073:31)
>   	at Generator.next (<anonymous>)
>   	at fulfilled (c:\Users\me\GitRepos\vscode-extension-for-zowe\packages\zowe-explorer\out\src\extension.js:189011:58)] {
>     mDetails: {
>       msg: 'Token auth error',
>       additionalDetails: 'Profile was found using token auth, please log in to continue.',
>       errorCode: '401'
>     },
>     mMessage: 'Token auth error'
>   },
>   label: 'dev.zosmf',
>   moreInfo: 'Token auth error'
> }
> ```
> 
> I am guessing that this is an issue with isUsingTokenAuth() picking up that I have used tokens in the past with certain profiles that I no longer have in my config. My current zowe.config.json only contains secure arrays with user and password.

Hey @adam-wolfe ! Could you provide the replication steps for this issue? i'm not sure if Zowe Explorer has a profile somehow cached here or what specifically the problem is 🤔

**Comment by adam-wolfe** [comment]
After pulling again and rebuilding, things seem to be working normally 🤷‍♂️. Not sure what happened there.

**Comment by rudyflores** [comment]
@traeok pushed the fix for it :) should be all good to go now

**Comment by traeok** [comment]
> pushed the fix for it :) should be all good to go now

Awesome, thanks @rudyflores!

**Review by traeok — DISMISSED** [review]

**Comment by rudyflores** [comment]
Saw this issue if logging out with an expanded session with token auth: might be after merging a couple changes after my token fix

Screenshot 2023-09-14 at 9 24 04 AM

**Comment by JillieBeanSim** [comment]
@rudyflores see [comment](https://github.com/zowe/vscode-extension-for-zowe/pull/2461#pullrequestreview-1624845023) in other fix that came in about a check that was introduced that cannot be that way, we cannot stop unverified profiles from doing actions.  The only difference between unverified and valid/invalid is the check for if server is running before doing an action, unverified profiles do not run that check.

**Comment by JillieBeanSim** [comment]
> > @rudyflores see [comment](https://github.com/zowe/vscode-extension-for-zowe/pull/2461#pullrequestreview-1624845023) in other fix that came in about a check that was introduced that cannot be that way, we cannot stop unverified profiles from doing actions. The only difference between unverified and valid/invalid is the check for if server is running before doing an action, unverified profiles do not run that check.
> 
> So to make sure I follow, we are missing a check for unverified profiles and that's why this error is happening?

@rudyflores you had added a check for unverified profiles and returned an empty array if the profile was unverified, we shouldn't do that so it was taken out

**Comment by rudyflores** [comment]
@traeok @t1m0thyj @JillieBeanSim Added a fix for the issue from my screenshot, seems like the issue was that for all views when logging out a function was called in their respective APIs and was trying to access a session object that didn't exist, so I added the proper checks before doing so which will now cause the old error handling from my PR to occur. To test, just log in with token auth, perform a filter search in any of the 3 views, and once the folder expands, log out and you will see the prompt at the bottom right appear (like in my PR)

**Review by traeok — CHANGES_REQUESTED** [review]
I'm running into an issue related to token auth: if the base profile in my Zowe config has a `tokenValue` specified in the secure array, I cannot execute a search with the direct-to-service profiles defined in the same layer. An info message appears with "Your connection is no longer active" when trying to search, and the search will not execute. However, searching works fine with a config that doesn't have `tokenValue` defined in its base profile secure array.

I think this is related to the checks added for secure values. We'll probably need to add a check to see if the current profile has a `user` and `password` defined in its secure array, and if so, disregard the token info from the base profile.

<details>
<summary>
Screenshot
</summary>


</details>

**Comment by rudyflores** [comment]
Should be all fixed now, only issue is for new profiles without `user` and `password` properties in the secure array since we don't have a way to select basic auth and token auth. But this should be addressed in Timothy's PR https://github.com/zowe/vscode-extension-for-zowe/pull/2445 which should have this good to go :)

**Review by traeok — DISMISSED** [review]
Direct-to-service and API ML profiles are now working as intended, as of 8afd2c1. LGTM

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 1]
```diff
@@ -142,7 +142,8 @@ export async function isUsingTokenAuth(profileName: string): Promise<boolean> {
     const secureProfileProps = await Profiles.getInstance().getSecurePropsForProfile(profileName);
     const secureBaseProfileProps = await Profiles.getInstance().getSecurePropsForProfile(baseProfile?.name);
     if (isUsingZosmf && baseProfile) {
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:144]: This `if(iszosmf && baseProfile)` will cause issues if another profile type is registered with API ML and using baseProfile to hold the "tokenType": "apimlAuthenticationToken" with tokenValue.

Comment (JillieBeanSim) [reply packages/zowe-explorer/src/utils/ProfilesUtils.ts:144]: could possibly simplify the check for all profile types, idk still testing it.
```
const baseProfile = Profiles.getInstance().getDefaultProfile("base");
const secureProfileProps = await Profiles.getInstance().getSecurePropsForProfile(profileName);
const secureBaseProfileProps = await Profiles.getInstance().getSecurePropsForProfile(baseProfile?.name);
const profileUsesBasicAuth = secureProfileProps.includes("user") && secureProfileProps.includes("password");
return (secureProfileProps.includes("tokenValue") || secureBaseProfileProps.includes("tokenValue")) && !profileUsesBasicAuth;
```

**Review by traeok — APPROVED** [review]

**Review by rudyflores — APPROVED** [review]
Looks good to me! I did notice an issue with creds not being up to date accross views for basic auth that wasn't on 2.10 so I filed an issue we can address on a potential patch release since it has a workaround for it: https://github.com/zowe/vscode-extension-for-zowe/issues/2465

**Pull request merged**